core/main/src/broker/broker_utils.rs

```diff
@@ -123,4 +159,173 @@ impl BrokerUtils {
                 .into()),
         }
     }
+
+    // async fn register_custom_callback(
+    //     broker: &ThunderBroker,
+    //     request_id: u64,
+    // ) -> tokio::sync::mpsc::Receiver<BrokerOutput> {
+    //     let (response_tx, response_rx) = mpsc::channel(10);
+    //     // Register custom callback to handle the response
+    //     broker
+    //         .register_custom_callback(
+    //             request_id,
+    //             BrokerCallback {
+    //                 sender: response_tx,
+    //             },
+    //         )
+    //         .await;
+    //     response_rx
+    // }
+
+    // fn get_next_id() -> u64 {
+    //     ATOMIC_ID.fetch_add(1, Ordering::SeqCst)
+    // }
+
+    async fn send_thunder_request(
+        &self,
+        ws_tx: &Arc<Mutex<SplitSink<WebSocketStream<TcpStream>, Message>>>,
+        request: &str,
+    ) -> Result<(), ()> {
+        let mut ws_tx = ws_tx.lock().await;
+        if let Err(e) = ws_tx.feed(Message::Text(request.to_string())).await {
+            error!("Failed to send Thunder request: {}", e);
+            return Err(());
+        }
+        if let Err(e) = ws_tx.flush().await {
+            error!("Failed to flush Thunder request: {}", e);
+            return Err(());
+        }
+        Ok(())
+    }
+
+    pub async fn handle_subscribe(
+        broker_request: BrokerRequest,
+        connection_req: BrokerConnectRequest,
+        callback: BrokerCallback,
+    ) {
+        let mut requests = Vec::new();
+        let endpoint = connection_req.endpoint.clone();
+        let (response_tx, _response_rx) = mpsc::channel::<BrokerRequest>(10);
+        let (c_tx, _c_tr) = mpsc::channel(2);
+
+        let subscription_map: Arc<RwLock<BrokerSubMap>> =
+            Arc::new(RwLock::new(connection_req.sub_map.clone()));
+
+        let alias = broker_request.rule.alias.clone();
+        let (callsign, method) = get_callsign_and_method_from_alias(&alias);
+        let method = method.clone();
+
+        let broker_sender = BrokerSender {
+            sender: response_tx,
+        };
+        let cleaner = BrokerCleaner {
+            cleaner: Some(c_tx.clone()),
+        };
+
+        let thunder_broker = ThunderBroker::new(broker_sender, subscription_map, cleaner, callback);
```

Comment:
Will this result in creating a new instance of ThunderBroker? There should only be a single instance of ThunderBroker in RippleMain. We need to use that existing instance to send the subscription request with a custom callback for receiving event notifications.